src/theory/ext_theory.cpp

```diff
@@ -245,15 +246,24 @@ bool ExtTheory::doInferencesInternal(int effort,
           {
             processed = true;
             markReduced(terms[i]);
+            // We have exp[i] => terms[i] = sr, convert this to a clause.
+            // This ensures the proof infrastructure can process this as a
+            // normal theory lemma.
             Node eq = terms[i].eqNode(sr);
-            Node expn =
-                exp[i].size() > 1
-                    ? NodeManager::currentNM()->mkNode(kind::AND, exp[i])
-                    : (exp[i].size() == 1 ? exp[i][0] : d_true);
+            Node lem = eq;
+            if (!exp[i].empty())
+            {
+              std::vector<Node> eei;
+              for (const Node& e : exp[i])
+              {
+                eei.push_back(e.negate());
+              }
+              eei.push_back(eq);
+              lem = nm->mkNode(kind::OR, eei);
```

Comment:
Suggestion: could use a NodeBuilder here to be slightly more efficient.

Reply:
This is not performance critical code. I prefer using vectors for cases like this since it makes the code more flexible.